homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -330,9 +332,33 @@ async def _async_replace_device(self, replace_device):
             if new_entity_id is not None:
                 entity_migration_map[new_entity_id] = entry
 
+        @callback
+        def _handle_state_change(entity_id: str, old_state, new_state):
+            # Wait for entities to finish cleanup
+            if new_state is None and entity_id in pending_entities:
+                pending_entities.remove(entity_id)
+            if not pending_entities:
+                wait_for_entities.set()
```

Comment:
This looks odd. wait_for_entities is not set at this point.

Reply:
Fixed in 041cc6e

Reply:
Seems that work. I would have thought you had to create the wait_for_entities variable before you defined the function at all for it to capture that variable. Seems that is not required in python.